11_TableMock/readme_es.md

```diff
@@ -0,0 +1,224 @@
+```
+
+- Definimos una api falsa (para hacerlo simple vamos a hacerlo síncrono) en _src/api/memberAPI.ts_:
+
+ _src/api/memberAPI.ts_
+
+ ```javascript
+import {MemberEntity} from '../model/member';
+import MembersMockData from './memberMockData';
+
+// Sync mock data API, inspired from:
+// https://gist.github.com/coryhouse/fd6232f95f9d601158e4
+class MemberAPI {
+  //This would be performed on the server in a real app. Just stubbing in.
+  private _clone (item) {
+  	return JSON.parse(JSON.stringify(item)); //return cloned copy so that the item is passed by value instead of by reference
+  };
+
+  // Just return a copy of the mock data
+  getAllMembers() : Array<MemberEntity> {
+		return this._clone(MembersMockData);
+	}
+}
+
+export const memberAPI = new MemberAPI();
+```
+
+- Ahora es momento de saltar dentro de una parte interesante, vamos a borrar _hello.tsx_ y _nameEdit.tsx_.
+
+- Vamos a crear un componente sin estado que mostrara una simple _memberRow.tsx_ fila.
+
+_src/memberRow.tsx_
+
+```javascript
+import * as React from 'react';
+import {MemberEntity} from './model/member';
+
+export const MemberRow = (props: {member : MemberEntity}) =>
+       <tr>
+         <td>
+           <img src={props.member.avatar_url} style ={{maxWidth: '10rem'}}/>
+         </td>
+         <td>
+           <span>{props.member.id}</span>
+         </td>
+         <td>
+           <span>{props.member.login}</span>
+         </td>
+       </tr>
+```
+
+> No podemos usar max-widh en el parametro style. Tenemos que escribir 'maxWidth' en los componentes de react.
+
+- Luego vamos a implementar un componente que mostrará una lista de miembros (y usaremos las filas), _membersTable.tsx_:
+
+_./src/membersTable.tsx_
+
+```javascript
+import * as React from 'react';
+import {MemberEntity} from './model/member';
+import {memberAPI} from './api/memberAPI';
+import {MemberRow} from './memberRow';
+
+interface Props {
+}
+
+// We define members as a state (the compoment holding this will be a container
+// component)
+interface State {
+  members : MemberEntity[]
+}
+
+// Nice tsx guide: https://github.com/Microsoft/TypeScript/wiki/JSX
+export class MembersTableComponent extends React.Component<Props, State> {
+
+  constructor(props : Props){
+        super(props);
+        // set initial state
+        this.state = {members: []};
+  }
+
+   // Standard react lifecycle function:
+   // https://facebook.github.io/react/docs/component-specs.html
+   public componentDidMount() {
+     this.setState({members: memberAPI.getAllMembers()})
+   }
+
+   public render() {
+
+       return (
+        <div className="row">
+          <h2> Members Page</h2>
+          <table className="table">
+            <thead>
+              <tr>
+                <th>
+                  Avatar
+                </th>
+                <th>
+                  Id
+                </th>
+                <th>
+                  Name
+                </th>
+              </tr>
```

Comment:
There's a bug here, we need thead, tr... This bug comes from long time ago,if you run the projecto and show the console log you will see it, can you jump into that and fix it? Thanks

Reply:
In spanish and english sample we have thead, tr tag.
![algo](https://user-images.githubusercontent.com/14540103/47620979-1a6b1900-daf1-11e8-9f88-491ee68dc216.png)

Reply:
Yes and it's wrong on both, check how html tables work and check whats wrong, thanks